src/application.c

```diff
@@ -282,6 +282,37 @@ void save_clicked_handler(GtkWidget *widget, struct swappy_state *state) {
   save_state_to_file_or_folder(state, NULL);
 }
 
+void save_as_clicked_handler(GtkWidget *widget, struct swappy_state *state) {
+  gchar *filename_suggestion;
+  GtkWidget *dialog;
+  GtkFileChooser *chooser;
+  GtkFileChooserAction action = GTK_FILE_CHOOSER_ACTION_SAVE;
+  gint res;
+
+  commit_state(state);
+  filename_suggestion = format_filename(state->config->save_filename_format);
+
+  dialog = gtk_file_chooser_dialog_new(NULL, state->ui->window, action,
+                                       ("_Cancel"), GTK_RESPONSE_CANCEL,
```

Comment:
Is there a reason why you're putting the `"_Cancel"` and `"_Save"` between parenthesis?

Regarding translating those, I'm not sure. The [gtk doc](https://docs.gtk.org/gtk3/const.STOCK_SAVE.html) seems to indicate what you are doing is valid. But I don't know if it means it's properly translated and using the base gtk `po` files. I'd need to research.

Reply:
I've switched to French, and the Close / Save do not seem to be translated. You'll have to use `gettext` I guess? I don't know if the ninja target will be smart enough to catch those. (I don't think there any translation done in the `.c` files. It's only in the `ui` files for now.)

Good luck. :)